CRM/MembershipExtras/Job/OfflineAutoRenewal/MultipleInstallmentPlan.php

```diff
@@ -131,6 +132,63 @@ private function createRecurringContribution() {
     $this->newRecurringContributionID = $newRecurringContribution['id'];
   }
 
+  /**
+   * Calculates the new period's start date from the largest membership end date
+   * of the previous period.
+   *
+   * @return string
+   *   The new period's start date.
+   * @throws \Exception
+   */
+  private function calculateNewPeriodStartDate() {
+    $latestDate = NULL;
+    $currentPeriodLines = $this->getRecurringContributionLineItemsToBeRenewed($this->currentRecurContributionID);
```

Comment:
I could be mistaken , but I see that getRecurringContributionLineItemsToBeRenewed() is called twice with the same recur contributions Id passed to it, so maybe its result can be cached and reused ? though would worth confirming first that it is the case